src/contributing/contributing_to_niwrap.md

```diff
@@ -1 +1,471 @@
+
+```json
+// Original descriptor (simplified)
+{
+  "name": "bet",
+  "inputs": [
+    {
+      "id": "fractional_intensity",
+      "name": "Fractional intensity threshold",
+      "type": "Number",
+      "integer": true,
+      "minimum": 0,
+      "maximum": 1
+    }
+  ]
+}
+```
+
+To fix this, you'd change the descriptor to:
+
+```json
+// Fixed descriptor
+{
+  "name": "bet",
+  "inputs": [
+    {
+      "id": "fractional_intensity",
+      "name": "Fractional intensity threshold",
+      "type": "Number",
+      "integer": false,
+      "minimum": 0,
+      "maximum": 1
+    }
+  ]
+}
+```
+
+### Adding a Missing Parameter
+
+If a tool has a parameter that isn't exposed in NiWrap, you can add it to the descriptor.
+
+#### Example: Adding a Missing Flag
+
+Suppose FSL's FAST tool has a `-N` flag for no bias field correction, but it's missing from the descriptor:
+
+```json
+// Original descriptor (simplified)
+{
+  "name": "fast",
+  "inputs": [
+    // existing parameters
+  ]
+}
+```
+
+You would add the new parameter:
+
+```json
+// Updated descriptor
+{
+  "name": "fast",
+  "inputs": [
+    // existing parameters
+    {
+      "id": "no_bias_field_correction",
+      "name": "No bias field correction",
+      "type": "Flag",
+      "command-line-flag": "-N",
+      "description": "Turns off bias field correction"
+    }
+  ]
+}
+```
+
+### Creating a New Descriptor
+
+If you want to add support for a completely new tool, you'll need to create a new descriptor from scratch.
+
+#### Example: Creating a New Descriptor
+
+Here's a simplified example of creating a descriptor for a fictional tool called `brainanalyze`:
+
+```json
+{
+  "name": "brainanalyze",
+  "tool-version": "1.0.0",
+  "description": "Analyzes brain structures in neuroimaging data",
+  "command-line": "brainanalyze [INPUT] [OUTPUT] [OPTIONS]",
+  "container-image": {
+    "image": "neuroimaging/brainanalyze:1.0.0",
+    "type": "docker"
+  },
+  "inputs": [
+    {
+      "id": "input_file",
+      "name": "Input file",
+      "type": "File",
+      "value-key": "[INPUT]",
+      "description": "Input neuroimaging file (NIFTI format)",
+      "optional": false
+    },
+    {
+      "id": "output_file",
+      "name": "Output file",
+      "type": "String",
+      "value-key": "[OUTPUT]",
+      "description": "Output file path",
+      "optional": false
+    },
+    {
+      "id": "smoothing",
+      "name": "Smoothing factor",
+      "type": "Number",
+      "value-key": "[OPTIONS]",
+      "command-line-flag": "-s",
+      "integer": false,
+      "minimum": 0,
+      "maximum": 10,
+      "description": "Smoothing factor (0-10)",
+      "optional": true,
+      "default-value": 2.5
+    }
+  ],
+  "output-files": [
+    {
+      "id": "output",
+      "name": "Output file",
+      "path-template": "[OUTPUT]",
+      "description": "Output analysis file"
+    }
+  ]
+}
+```
+
```

Comment:
I wonder if there should be an "advanced" part about the nesting / sub-commands, but that can also come later.

Reply:
There is a placeholder for this. It will be a whole chapter in `boutiques_guide/`